frontend/e2e/$lang+/_public/apply/adult/_adult-flow.spec.ts

```diff
@@ -0,0 +1,234 @@
+import type { Page } from '@playwright/test';
+import { expect, test } from '@playwright/test';
+
+// Reusable funtion to check empty input
+async function hasError(page: Page, error: string) {
+  await page.getByRole('button', { name: 'Continue' }).click();
+  await expect(page.getByRole('link', { name: error })).toBeVisible();
+}
+
+// Reusable function to fill out date of birth
+async function fillOutDOB(page: Page, year: string) {
+  await page.getByRole('button', { name: 'Continue' }).click();
+  await page.getByRole('combobox', { name: 'Month' }).selectOption('01');
+  await page.getByRole('textbox', { name: 'Day (DD)' }).fill('1');
+  await page.getByRole('textbox', { name: 'Year (YYYY)' }).fill(year);
+}
+
+// Reusable function to fill out address
+async function fillOutAddress(page: Page, group: string, address: string, unit: string, country: string, province: string, city: string, postalCode: string) {
+  await page.getByRole('group', { name: group }).getByRole('textbox', { name: 'Address', exact: true }).fill(address);
+  await page.getByRole('group', { name: group }).getByRole('textbox', { name: 'Apartment, suite, etc. (optional)', exact: true }).fill(unit);
+  await page.getByRole('group', { name: group }).getByRole('combobox', { name: 'Country', exact: true }).selectOption(country);
+  await page.getByRole('group', { name: group }).getByRole('combobox', { name: 'Province, territory, state, or region', exact: true }).selectOption(province);
+  await page.getByRole('group', { name: group }).getByRole('textbox', { name: 'City or town', exact: true }).fill(city);
+
+  //invalid postal code
+  await page.getByRole('group', { name: group }).getByRole('textbox', { name: 'Postal code or ZIP code', exact: true }).fill('12345678');
+  await hasError(page, 'Enter mailing address postal code in the correct format, such as A1A 1A1');
+
+  await page.getByRole('group', { name: group }).getByRole('textbox', { name: 'Postal code or ZIP code', exact: true }).fill(postalCode);
+}
+
+test.describe('Adult flow', () => {
+  test('should complete application as adult applicant', async ({ page }) => {
+    await test.step('Terms and condition page', async () => {
+      await page.goto('/en/apply');
```

Comment:
Move this to his own step named `navigate to online application`. I think the step name/title should represent the step actions that is happening. Ex.: `Terms and condition page` => `agree and continue to the terms and coniditions`, `choose application type applying for myself and continue`. I might be wrong.